h2/src/main/org/h2/value/Transfer.java

```diff
@@ -594,6 +599,11 @@ public Value readValue() throws IOException {
             return ValueDouble.get(readDouble());
         case Value.FLOAT:
             return ValueFloat.get(readFloat());
+        case Value.ENUM: {
+            final int ordinal = readInt();
+            final String label = readString();
+            return ValueEnumBase.get(label, ordinal);
```

Comment:
Hey @grandinj I've put the solution you described here in place - different `ValueEnum` for the client and server.